src/flag_gems/ops/hstack.py

```diff
@@ -0,0 +1,57 @@
+import itertools
+import logging
+from typing import List, Tuple, Union
+
+import torch
+import triton
+
+from ..utils import pointwise_dynamic
+from ..utils.tensor_wrapper import StridedBuffer
+
+
+@pointwise_dynamic(is_tensor=[True], promotion_methods=[(0, "DEFAULT")])
+@triton.jit
+def copy_func(x):
+    return x
+
+
+def hstack(
+    tensors: Union[Tuple[torch.Tensor, ...], List[torch.Tensor]]
+) -> torch.Tensor:
+    logging.debug("GEMS HSTACK")
+
+    if len(tensors) == 0:
+        raise RuntimeError("hstack expected a non-empty TensorList")
+
+    inp_shapes = [list(_.shape) for _ in tensors]
+    inp0_shape = inp_shapes[0]
+    out_shape = list(inp0_shape)
+    if len(inp0_shape) == 1:
+        dim = 0
+    else:
+        dim = 1
+        for tensor_num, inp_shape in enumerate(inp_shapes[1:]):
+            for i in range(len(inp_shape)):
+                if i != dim and inp_shape[i] != inp0_shape[i]:
```

Comment:
We should ensure that all tensors have the same ndim, or else inp0_shape[i] may cause an `out of range` error.

torch.hstack would raise an error `Tensors must have same number of dimensions: got {i} and {j}` if any two of the input tensors have the different ndim. with an only exception that tensor of shape `()` would be preprocessed to `(1, )`, (I think this is a not that strict behavior).

If we are going to follow what torch.hsatck does, the following preprocessing or checking should be done beforehand.
1. preprocess 0-d tensors, view them as 1d tensors;
2. check all tensors have the same ndim.

Reply:
Done